design/kopia.md

```diff
@@ -0,0 +1,471 @@
+# Kopia Integration
+
+<!-- toc -->
+- [Custom Resource Definition](#custom-resource-definition)
+- [Repository Lifecycle](#repository-lifecycle)
+  - [Server-Side Setup](#server-side-setup)
+  - [Client-Side Setup](#client-side-setup)
+- [Server Access Users Management](#server-access-users-management)
+  - [Predefined Users List](#predefined-users-list)
+  - [Dynamic Users List](#dynamic-users-list)
+- [Integration Modes](#integration-modes)
+  - [Global Mode](#global-mode)
+  - [Namespace Mode](#namespace-mode)
+  - [Migration Between Modes](#migration-between-modes)
+- [Secrets Management](#secrets-management)
+<!-- /toc -->
+
+This document provides design information on concepts and details relevant
+to the Kanister's integration with [Kopia][9].
+
+## Custom Resource Definition
+
+This design proposes a new Custom Resource Definition (CRD) named `Repository`,
+to represent a [Kopia repository][1]. It is a namespace-scoped resource, owned
+and managed by a new `RepositoryController` controller in the `kanister`
+namespace. The CRD offers a set of parameters to configure a Kopia repository.
+
+To limit the new controller's RBAC scope, all `Repository` resources are
+created in the `kanister` namespace.
+
+A sample `Repository` resource used to store snapshot artifacts of workloads
+running in the `monitoring` namespace looks like this:
+
+```yaml
+apiVersion: cr.kanister.io/v1alpha1
+kind: Repository
+metadata:
+  name: repository-monitoring
+  namespace: kanister
+  labels:
+    repo.kanister.io/target-namespace: monitoring
+spec:
+  # immutable
+  targetNamespace: monitoring
+  location:
+    # required
+    secretName: repository-monitoring-location
+  server:
```

Comment:
What about Repository path or `repoPath` field? Are we going to internally create it using a pre-defined format and the location? @pavannd1

Reply:
Depends if we want to give user the option to change it. To keep it simple, I think the controller should generate it, to avoid having the handle scenarios like what if user changed the path.